src/visualization/context/pagination.tsx

```diff
@@ -69,7 +69,7 @@ export const PaginationProvider: FC<PaginationProviderProps> = ({
     (page: number) => {
       setOffset(calcOffset(page, maxSize, total))
     },
-    [offset, maxSize, setOffset] // eslint-disable-line react-hooks/exhaustive-deps
+    [maxSize, setOffset, total]
```

Comment:
`offset` is not needed here since it's not used within the hook. `total` is needed and added to the dependency array.